lib/httplib/httplib.go

```diff
@@ -149,10 +150,16 @@ func ReadJSON(r *http.Request, val interface{}) error {
 // based on HTTP response code and HTTP body contents
 func ConvertResponse(re *roundtrip.Response, err error) (*roundtrip.Response, error) {
 	if err != nil {
-		if uerr, ok := err.(*url.Error); ok && uerr != nil && uerr.Err != nil {
-			return nil, trace.ConnectionProblem(uerr.Err, uerr.Error())
+		var uErr *url.Error
+		if errors.As(err, &uErr) && uErr.Err != nil {
```

Comment:
If `errors.As()` returns true, we have a `*url.Error` and shouldn't proceed as if we don't.

Reply:
> proceed as if we don't

Do you mean we don't need to check `uErr.Err != nil` after?

Reply:
Actually, I misread -- the error gets returned on 167 regardless.